pkg/providers/instance/resperrorhandlers.go

```diff
@@ -143,24 +144,30 @@ func handleSKUNotAvailableError(ctx context.Context, provider *DefaultProvider,
 		capacityType)
 }
 
-func handleZonalAllocationFailureError(ctx context.Context, provider *DefaultProvider, instanceType *corecloudprovider.InstanceType, zone, capacityType string, err error) error {
+// For zonal allocation failure, we will mark all instance types from this SKU family that have >= CPU count as the one that hit the error in this zone
+func handleZonalAllocationFailureError(ctx context.Context, provider *DefaultProvider, sku *skewer.SKU, instanceType *corecloudprovider.InstanceType, zone, capacityType string, err error) error {
 	log.FromContext(ctx).Error(err, "zonal allocation failure", "instance-type", instanceType.Name, "zone", zone)
-	provider.unavailableOfferings.MarkUnavailableWithTTL(ctx, ZonalAllocationFailureReason, instanceType.Name, zone, karpv1.CapacityTypeOnDemand, AllocationFailureTTL)
-	provider.unavailableOfferings.MarkUnavailableWithTTL(ctx, ZonalAllocationFailureReason, instanceType.Name, zone, karpv1.CapacityTypeSpot, AllocationFailureTTL)
+	vCPU, err := sku.VCPU() // versionedSKUFamily e.g. "N4" for "NV8as_v4"
+	if err != nil {
+		// default to 0 if we can't determine VCPU count, this shouldn't happen as long as data in skewer.SKU is correct
+		vCPU = 0
+	}
+	provider.unavailableOfferings.MarkFamilyUnavailableWithTTL(ctx, sku.GetFamilyName(), zone, karpv1.CapacityTypeOnDemand, vCPU, AllocationFailureTTL)
```

Comment:
One thing that I notice on our calls here.

There's only a _single_ call to `unavailableOfferings` with the default TTL. There's 5 with `AllocationFailureTTL`, and one with `skuNotAvailableTTL` (which is a bit dynamic).

This makes me wonder if we just want the unavailableOfferings interface to always require a TTL, since there seem to be a couple different options and most aren't using the default.

We _could_ make the default `AllocationFailureTTL` since that's used most often, which would be another angle that would be IMO better than what we have now.

Reply:
True, it is kind of all over the place now. I think we could maybe address it by adding options parameters like you mentioned, but I'd prefer to do it in a separate PR, this one is pretty bloated already.

Reply:
Agree doing it in a separate PR would be better.